File: sportsbook-app/betting-service/src/main/java/com/sportsbook/bettingservice/kafka/BetEventProducer.java
package com.sportsbook.bettingservice.kafka;

import com.sportsbook.bettingservice.domain.Bet;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class BetEventProducer {

    private final KafkaTemplate<String, BetPlacedEvent> kafkaTemplate;

    @Value("${app.kafka.topics.bet-placed}")
    private String betPlacedTopic;

    public void publishBetPlaced(BetPlacedEvent event) {
        // use bet id as the key if you want partition affinity
        kafkaTemplate.send(betPlacedTopic, event.getId(), event);
    }
}

File: sportsbook-app/betting-service/src/main/kotlin/com/sportsbook/bettingservice/kafka/BetEventProducer.kt
package com.sportsbook.bettingservice.kafka

import org.springframework.beans.factory.annotation.Value
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component

@Component
class BetEventProducer(
    private val kafkaTemplate: KafkaTemplate<String, BetPlacedEvent>,
    @Value("\${app.kafka.topics.bet-placed}") private val betPlacedTopic: String,
) {

    fun publishBetPlaced(event: BetPlacedEvent) {
        val key = requireNotNull(event.id) { "BetPlacedEvent.id is required" }
        kafkaTemplate.send(betPlacedTopic, key, event)
    }
}
